File: .claude/worktrees/localize-strings/iosApp/eSklepios/Components/PractitionerCard.swift
import SwiftUI

struct PractitionerCardData {
    let id: String
    let firstName: String
    let lastName: String
    let specialty: String
    let clinicName: String
    let address: String
    let city: String
    let acceptingNewPatients: Bool
    let isFavorite: Bool
    let nextSlots: [String]

    var fullName: String { "Dr. \(firstName) \(lastName)" }
    var initials: String {
        let f = firstName.first.map(String.init) ?? ""
        let l = lastName.first.map(String.init) ?? ""
        return "\(f)\(l)".uppercased()
    }
}

struct PractitionerCard: View {
    let data: PractitionerCardData
    let onSeeProfile: () -> Void
    let onBook: (String) -> Void
    let onFavorite: () -> Void

    var body: some View {
        AppCard {
            VStack(alignment: .leading, spacing: Dimens.paddingM) {
                // Header row
                HStack(alignment: .top, spacing: Dimens.paddingM) {
                    AvatarCircle(initials: data.initials, size: Dimens.avatarLg)

                    VStack(alignment: .leading, spacing: 3) {
                        Text(data.fullName)
                            .font(.system(size: 15, weight: .bold))
                            .foregroundColor(.appTextPrimary)
                        Text(data.specialty)
                            .font(.system(size: 13))
                            .foregroundColor(.appPrimary)
                        Text(data.clinicName)
                            .font(.system(size: 12))
                            .foregroundColor(.appTextSecondary)
                    }

                    Spacer()

                    VStack(alignment: .trailing, spacing: 6) {
                        Button(action: onFavorite) {
                            Image(systemName: data.isFavorite ? "heart.fill" : "heart")
                                .font(.system(size: 18))
                                .foregroundColor(data.isFavorite ? Color(hex: "FF6B6B") : Color.appTextSecondary)
                        }

                        if data.acceptingNewPatients {
                            Text("Accepting")
                                .font(.system(size: 10, weight: .semibold))
                                .padding(.horizontal, 7)
                                .padding(.vertical, 3)
                                .background(Color.appSuccessBg)
                                .foregroundColor(.appSuccess)
                                .cornerRadius(Dimens.radiusPill)
                        }
                    }
                }

                // Address row
                HStack(spacing: 4) {
                    Image(systemName: "mappin")
                        .font(.system(size: 12))
                        .foregroundColor(.appTextSecondary)
                    Text("\(data.address), \(data.city)")
                        .font(.system(size: 12))
                        .foregroundColor(.appTextSecondary)
                        .lineLimit(1)
                }

                Divider()
                    .background(Color.appTextHint.opacity(0.3))

                // Available slots
                if !data.nextSlots.isEmpty {
                    VStack(alignment: .leading, spacing: 6) {
                        Text("Available slots")
                            .font(.system(size: 11, weight: .semibold))
                            .foregroundColor(.appTextSecondary)

                        ScrollView(.horizontal, showsIndicators: false) {
                            HStack(spacing: 8) {
                                ForEach(data.nextSlots.prefix(4), id: \.self) { slot in
                                    Button(action: { onBook(slot) }) {
                                        Text(slot)
                                            .font(.system(size: 12, weight: .medium))
                                            .padding(.horizontal, 10)
                                            .padding(.vertical, 5)
                                            .background(Color.appPrimaryLight)
                                            .foregroundColor(.appPrimary)
                                            .cornerRadius(Dimens.radiusSm)
                                            .overlay(
                                                RoundedRectangle(cornerRadius: Dimens.radiusSm)
                                                    .stroke(Color.appPrimary.opacity(0.3), lineWidth: 1)
                                            )
                                    }
                                }
                            }
                        }
                    }
                }

                // Action buttons
                HStack(spacing: Dimens.paddingM) {
                    SecondaryButton(title: "See Profile", action: onSeeProfile)
                    PrimaryButton(title: "Book", action: {
                        if let slot = data.nextSlots.first { onBook(slot) }
                    })
                }
            }
        }
    }
}

#Preview {
    PractitionerCard(
        data: PractitionerCardData(
            id: "1",
            firstName: "Sarah",
            lastName: "Johnson",
            specialty: "Cardiologist",
            clinicName: "Heart Clinic",
            address: "12 Rue de la Santé",
            city: "Luxembourg",
            acceptingNewPatients: true,
            isFavorite: false,
            nextSlots: ["10:00", "11:30", "14:00"]
        ),
        onSeeProfile: {},
        onBook: { _ in },
        onFavorite: {}
    )
    .padding()
    .background(Color.appBackground)
}

File: .claude/worktrees/no-hardcoded-strings/androidApp/src/main/kotlin/lu/esklepios/app/ui/components/PractitionerCard.kt
package lu.esklepios.app.ui.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.LocationOn
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import lu.esklepios.app.R
import lu.esklepios.app.ui.theme.*

data class PractitionerUiModel(
    val id: String,
    val firstName: String,
    val lastName: String,
    val specialty: String,
    val clinic: String,
    val address: String,
    val isAcceptingNewPatients: Boolean,
    val availableSlots: List<SlotDayUiModel>
) {
    val initials: String get() = "${firstName.firstOrNull() ?: ""}${lastName.firstOrNull() ?: ""}"
    val fullName: String get() = "Dr. $firstName $lastName"
}

data class SlotDayUiModel(
    val dayLabel: String,
    val slots: List<String>
)

@Composable
fun PractitionerCard(
    practitioner: PractitionerUiModel,
    onSeeProfile: () -> Unit,
    onBook: (slotId: String) -> Unit,
    modifier: Modifier = Modifier
) {
    AppCard(modifier = modifier.fillMaxWidth()) {
        Column(modifier = Modifier.padding(Dimens.paddingL)) {
            // Header row: avatar + name + badge
            Row(verticalAlignment = Alignment.Top) {
                AvatarCircle(
                    initials = practitioner.initials,
                    size = Dimens.avatarSizeLg,
                    fontSize = 20.sp
                )
                Spacer(modifier = Modifier.width(Dimens.paddingM))
                Column(modifier = Modifier.weight(1f)) {
                    Text(
                        text = practitioner.fullName,
                        style = MaterialTheme.typography.titleMedium.copy(
                            color = TextPrimary,
                            fontWeight = FontWeight.Bold
                        )
                    )
                    Text(
                        text = practitioner.specialty,
                        style = MaterialTheme.typography.bodySmall.copy(color = Primary)
                    )
                    Text(
                        text = practitioner.clinic,
                        style = MaterialTheme.typography.bodySmall.copy(color = TextSecondary)
                    )
                }
                if (practitioner.isAcceptingNewPatients) {
                    Box(
                        modifier = Modifier
                            .background(color = SuccessBg, shape = RoundedCornerShape(Dimens.radiusPill))
                            .padding(horizontal = Dimens.paddingS, vertical = 2.dp)
                    ) {
                        Text(
                            text = stringResource(R.string.label_accepting_patients),
                            style = MaterialTheme.typography.labelSmall.copy(
                                color = Success,
                                fontSize = 10.sp,
                                fontWeight = FontWeight.SemiBold
                            )
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.height(Dimens.paddingM))

            // Address row
            Row(verticalAlignment = Alignment.CenterVertically) {
                Icon(
                    imageVector = Icons.Filled.LocationOn,
                    contentDescription = null,
                    tint = TextSecondary,
                    modifier = Modifier.size(Dimens.iconSizeSm)
                )
                Spacer(modifier = Modifier.width(Dimens.paddingXS))
                Text(
                    text = practitioner.address,
                    style = MaterialTheme.typography.bodySmall.copy(color = TextSecondary)
                )
            }

            Spacer(modifier = Modifier.height(Dimens.paddingM))
            Divider(color = BorderLight)
            Spacer(modifier = Modifier.height(Dimens.paddingM))

            // Available slots
            if (practitioner.availableSlots.isNotEmpty()) {
                Text(
                    text = stringResource(R.string.label_available_slots),
                    style = MaterialTheme.typography.labelSmall.copy(
                        color = TextSecondary,
                        fontWeight = FontWeight.SemiBold
                    )
                )
                Spacer(modifier = Modifier.height(Dimens.paddingS))
                LazyRow(
                    horizontalArrangement = Arrangement.spacedBy(Dimens.paddingM)
                ) {
                    items(practitioner.availableSlots) { daySlot ->
                        Column(
                            horizontalAlignment = Alignment.CenterHorizontally,
                            verticalArrangement = Arrangement.spacedBy(Dimens.paddingXS)
                        ) {
                            Text(
                                text = daySlot.dayLabel,
                                style = MaterialTheme.typography.labelSmall.copy(
                                    color = TextSecondary,
                                    fontSize = 10.sp
                                )
                            )
                            daySlot.slots.forEach { slot ->
                                Box(
                                    modifier = Modifier
                                        .clip(RoundedCornerShape(Dimens.radiusSm))
                                        .background(PrimaryLight)
                                        .border(
                                            width = 1.dp,
                                            color = Primary.copy(alpha = 0.3f),
                                            shape = RoundedCornerShape(Dimens.radiusSm)
                                        )
                                        .clickable { onBook(slot) }
                                        .padding(
                                            horizontal = Dimens.paddingS,
                                            vertical = Dimens.paddingXS
                                        ),
                                    contentAlignment = Alignment.Center
                                ) {
                                    Text(
                                        text = slot,
                                        style = MaterialTheme.typography.labelSmall.copy(
                                            color = Primary,
                                            fontSize = 11.sp,
                                            fontWeight = FontWeight.Medium
                                        )
                                    )
                                }
                            }
                        }
                    }
                }
                Spacer(modifier = Modifier.height(Dimens.paddingM))
            }

            // Action buttons
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(Dimens.paddingM)
            ) {
                SecondaryButton(
                    text = stringResource(R.string.action_see_profile),
                    onClick = onSeeProfile,
                    modifier = Modifier.weight(1f)
                )
                PrimaryButton(
                    text = stringResource(R.string.action_book),
                    onClick = { practitioner.availableSlots.firstOrNull()?.slots?.firstOrNull()?.let { onBook(it) } },
                    modifier = Modifier.weight(1f)
                )
            }
        }
    }
}
